lib/tic_tac_toe_api_web/controllers/page_controller.ex

```diff
@@ -5,4 +5,29 @@ defmodule TicTacToeApiWeb.PageController do
     board = %Board{}
     render(conn, "index.json", board: board)
   end
+
+  def status(conn, %{"spots" => spots, "next_player" => next_player, "current_player" => current_player}) do
+    player1 = %Player{symbol: next_player}
+    player2 = %Player{symbol: current_player}
+    spots_list = String.replace(spots, ~r/[][\",]/, "")
+                 |> String.split(" ")
+                 |> swap_symbol_with_player(player1, player2)
+    result = Status.result(spots_list, next_player, current_player)
+    render(conn, "status.json", result: result)
+  end
+
+  defp swap_symbol_with_player([], _player1, _player2), do: []
+
+  defp swap_symbol_with_player([head | tail], player1, player2) do
+    cond do
+      player1.symbol == head ->
+        [player1] ++ swap_symbol_with_player(tail, player1, player2)
+      player2.symbol == head ->
+        [player2] ++ swap_symbol_with_player(tail, player1, player2)
+      true ->
+        [head] ++ swap_symbol_with_player(tail, player1, player2)
+    end
```

Comment:
I don't know Elixir super well, but looking at this I would bet good money there is a more Elixir-y way to do this and simplify the code. Have a look at this blog post and see if you can't refactor this to use pattern-matching? https://nickjanetakis.com/blog/refactoring-elixir-code-if-cond-and-pattern-matching

Reply:
we tried to use pattern matching with the case statements as shown in the blog, but unfortunately in this case using `case...do` doesn't allow us to access the player structs in pattern matching so we had to just go with `cond...do`

Reply:
Ok, it took me awhile to try to figure out what is going on here, but I might be able to provide some context / advice to help clean this up. Let me know if any part of my analysis is incorrect (a lot of this is conjecture based on the limited code that I am seeing here, so it might be way off base). Long story short, fixing this one function should be part of a solution to fix a larger, underlying architectural concern, but it is not sufficient by itself.

The `swap_symbol_with_player` function is taking an array of marks and replacing it with an array of player structs. This array of player structs is used in the Elixir/Phoenix app (or so I assume).

When you get data from some endpoint, it comes back as raw JSON. I think(?) you are attempting to convert raw JSON into an Elixir struct using regular expressions (i.e. `spots_list = String.replace(spots, ~r/[][\",]/, "")`).

Thus, you end up with this weird situation - the frontend stores data in terms of "marks" (or so I assume) while the backend stores data in terms of player structs. So, how would we fix something like this? Of course you both already know because you attended the workshop last week :P 

1. Don't decode the JSON using regex. It is error-prone and I think you missed some edge cases in your regex expression. Instead, use a library like Jason.

2. Consider changing the frontend architecture to store player data in terms of player objects instead of marks. That way, you can send back structured data rather than marks (e.g. `"currentMarks": [{"player": "Player One", "space": 5, "symbol": "X"}]`)

3. Item number 2 might be too big of a jump right now or unrealistic in the time left in the sprint. Instead, create a new module with an adapter function that converts a list of marks to a list of players. That way, that conversion logic is isolated from your core business logic